### DISCUSSION TITLE
MQTT Payload Data Meaning - Before/After

### DISCUSSION THREAD

**Discussion by Archer36** [blakeblackshear/frigate#7194] [Ask A Question]
Hello, hopefully a quick question. Is there a place in the documentation which explains what the difference is between "before" and "after" in MQTT event messages? I see there is some examples where you can leverage before/after to generate an alert when someone is leaving a zone but I'm still not clear on what those two labels actually mean in the grand scheme.

Thanks!

**Comment by NickM-27** [comment]
Frigate only sends messages at certain points like when an object is first detected as a true positive, enters a zone, etc. So before is the state of the object before that moment and after is the state of the object after that moment

**Comment by danodemano** [comment]
I don't mean to revive an old question but I hope that's better than creating a new one. I'm still not clear on what "before" and "after" mean in terms of the MQTT messages. I'm working on custom alerting and up until this point have been using "before" but that doesn't seem to (always) match up with an event as tagged in Frigate. I'm trying to understand what's what and how to properly interpret these messages. Thanks so much!

**Reply by NickM-27** [reply]
It is what the state was right before and right after the point in time when something significant changed like the object entered a zone

**Reply by danodemano** [reply]
So this is probably a really, really dumb question. But what is the correct one to reference to correlate to an "event" in frigate? I feel like I have a disconnect there in my understanding. 

Also, thanks for the crazy-fast reply!

**Reply by NickM-27** [reply]
You really need to reference both of you don't want repeated notifications. In this case you'd check that before has_snapshot and has_clip are false and after has_snapshot or has_clip is true

**Reply by danodemano** [reply]
Ahh, fantastic! That would be the piece of data I was missing. Cheers!
